File: result/translate/starchat/java2cpp/c++/INTEGER_POSITIVE_VALUE_POSITIVE_NEGATIVE_VALUE_ARRAY.cpp
int f_gold ( int arr [ ], int n ) {
    unordered_map<int, int> hash;
    int maximum = 0;
    for (int i = 0; i < n; i++) {
        if (arr[i] < 0) hash[abs(arr[i])]--;
        else hash[abs(arr[i])]++;
        if (hash[abs(arr[i])] > maximum) maximum = hash[abs(arr[i])];
    }
    for (int i = 0; i < n; i++) if (hash[arr[i]] > 0) return arr[i];
    return -1;
}

File: result/translate/transcoder/java2python/python/INTEGER_POSITIVE_VALUE_POSITIVE_NEGATIVE_VALUE_ARRAY.py
def f_gold ( arr , n ) :
    hash = { }
    maximum = 0
    for i in range ( n ) :
        if arr [ i ] < 0 :
            hash [ abs ( arr [ i ] ) ] = ( hash [ abs ( arr [ i ] ) ] if i in hash else 0 ) - 1
        else :
            hash [ abs ( arr [ i ] ) ] = ( hash [ abs ( arr [ i ] ) ] if i in hash else 0 ) + 1
    for i in range ( n ) :
        if hash [ i ] > 0 :
            return arr [ i ]
    return - 1